### PR TITLE
Chore: remove currentFilename prop from Linter instances (refs #9161)

### PR DESCRIPTION
**Pull request by not-an-aardvark**
**What is the purpose of this pull request? (put an "X" next to item)**

[x] Other, please explain:

**What changes did you make? (Give an overview)**

This removes the undocumented `currentFilename` property from `Linter` instances.

This relies on https://github.com/eslint/eslint/pull/9218. The tests will pass after https://github.com/eslint/eslint/pull/9218 is merged.

**Is there anything you'd like reviewers to focus on?**

Nothing in particular

### CODE REVIEW

**Comment by not-an-aardvark** [comment]
¯\\\_(ツ)_/¯ I'm fine with it because we have the ["use at your own risk"](https://eslint.org/docs/developer-guide/nodejs-api) warning for undocumented APIs.

**Comment by platinumazure** [comment]
Possible compromise: Could we consider this an Update rather than a Chore and make it semver-minor?

**Comment by not-an-aardvark** [comment]
I'm fine with including this in a semver-minor release as a compromise, although I don't see how it would end up there based on our semver policy. To me, it seems like one of the following is true:

1. There are integration developers ignoring the "use at your own risk" warning, and we're willing to accomodate them to avoid breaking builds, so we consider it a breaking change, or:
1. (1) isn't true, so this is just an internal refactor, which is semver-patch.

Basically, I don't see how this fits under `Update:` in either case.
